pkg/ceph/osd/daemon.go

```diff
@@ -86,6 +91,21 @@ func Run(context *clusterd.Context, agent *OsdAgent) error {
 	return nil
 }
 
+// Set the hostname to the hostname of the node. We don't want the hostname set by the pod,
+// which is the name of the pod, which would change if the pod is re-created.
+func setHostName(context *clusterd.Context, nodeName string) error {
```

Comment:
For consistent terminology I'll change this to `setNodeName`. But under the covers it needs to call `hostname` to set the name since that is how the osd will pick up the name for the crush map.